src/core/mesh/qgsmesheditor.cpp

```diff
@@ -0,0 +1,1314 @@
+/***************************************************************************
+  qgsmesheditor.cpp - QgsMeshEditor
+
+ ---------------------
+ begin                : 8.6.2021
+ copyright            : (C) 2021 by Vincent Cloarec
+ email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsmesheditor.h"
+#include "qgsmeshdataprovider.h"
+#include "qgstriangularmesh.h"
+#include "qgsmeshlayer.h"
+#include "qgslogger.h"
+
+#include <poly2tri.h>
+
+#include <QSet>
+
+QgsMeshEditor::QgsMeshEditor( QgsMeshLayer *meshLayer ):
+  QObject( meshLayer )
+  , mMesh( meshLayer ? meshLayer->nativeMesh() : nullptr )
+  , mTriangularMesh( meshLayer ? meshLayer->triangularMeshByLodIndex( 0 ) : nullptr )
+  , mUndoStack( meshLayer ? meshLayer->undoStack() : nullptr )
+{}
+
+QgsMeshEditor::QgsMeshEditor( QgsMesh *nativeMesh, QgsTriangularMesh *triangularMesh, QObject *parent ):
+  QObject( parent )
+  , mMesh( nativeMesh )
+  , mTriangularMesh( triangularMesh )
+{
+  mUndoStack = new QUndoStack( this );
+}
+
+QgsTopologicMesh::TopologicError QgsMeshEditor::initialize()
+{
+  QgsTopologicMesh::TopologicError error;
+  mTopologicMesh = QgsTopologicMesh::createTopologicMesh( mMesh, error );
+  return error;
+}
+
+void QgsMeshEditor::applyEdition( QgsMeshEditor::Edition &edition )
```

Comment:
"Edition" is not used in English in this sense https://dictionary.cambridge.org/dictionary/english/edition

Better to use "Edit", e.g. applyEdit. Please fix in the all occurrences of "edition" in the code